ocular/src/chain/client/transactions.rs

```diff
@@ -0,0 +1,170 @@
+use crate::error::TransactionError;
+use cosmrs::{
+    bank::MsgSend,
+    crypto::{secp256k1::SigningKey, PublicKey},
+    staking::{MsgDelegate, MsgUndelegate},
+    tendermint::chain::Id,
+    tx::{self, Fee, Msg, SignDoc, SignerInfo},
+    AccountId, Coin,
+};
+use tendermint_rpc::endpoint::broadcast::tx_commit::Response;
+
+use super::ChainClient;
+
+/// Metadata wrapper for transactions
+#[derive(Clone, Debug)]
+pub struct TransactionMetadata {
+    pub chain_id: Id,
+    pub account_number: u64,
+    pub sequence_number: u64,
+    pub gas_limit: u64,
+    pub timeout_height: u16,
+    pub memo: String,
+}
```

Comment:
I'm betting we can build this behind the client API rather than requiring the user to make it?

Reply:
we can probably fill in chain id (and ~maybe~ sequence number, though I can imagine a use case or 2 where even that might want to be configurable), but other than that I think each of these fields the user might want to configure in one use case or another (e.g. different gas limits for different tx's, diff memos, diff timeout heights, etc)